### PR TITLE
Multi-read graph display, with more layout options

### PR DESCRIPTION
**Pull request by rlorigro**
Added ability to display graph starting from multiple reads and use multiple layout methods. Also removed unused display option for containment edges.

### CODE REVIEW

File: src/Assembler.hpp [hunk 1]
```diff
@@ -1067,12 +1067,23 @@ class shasta::Assembler :
...
+            bool allowCrossStrandEdges,
+            size_t maxTrim,         // Used to define containment.
+            double timeout,         // Or 0 for no timeout.
+            LocalReadGraph&);
+
+    // Create a local subgraph of the global read graph,
+    // starting at any number of  given vertexes and extending out to a specified
+    // distance (number of edges).
+    bool createLocalReadGraph(
+        vector<OrientedReadId> starts,
```
Comment (bagashe) [inline src/Assembler.hpp:1086]: This will `copy` the vector. Pass it as a reference instead? - `const vector<OrientedReadId>& starts`

Comment (rlorigro) [reply src/Assembler.hpp]: Yes, good catch, thanks

File: src/AssemblerHttpServer-ReadGraph.cpp [hunk 2]
```diff
@@ -31,18 +31,53 @@ void Assembler::exploreReadGraph(
 }
 
 
+bool parseCommaSeparatedReadIDs(string& commaSeparatedReadIDs, vector<OrientedReadId>& readIds, ostream& html){
```
Comment (paoloczi) [inline src/AssemblerHttpServer-ReadGraph.cpp:34]: Shasta uses naming conventions where, in a name containing an acronym, only the first letter is capitalized. So in the name of this function and the first argument `ID` should become `Id`. This is also consistent with `OrientedReadId` and `readIds`.

File: src/AssemblerHttpServer-ReadGraph.cpp [hunk 3]
```diff
@@ -31,18 +31,53 @@ void Assembler::exploreReadGraph(
...
+    readIds.clear();
+    string token;
+
+    for (auto& c: commaSeparatedReadIDs){
+        if (c == ','){
+            try {
+                OrientedReadId readID(token);
+                readIds.emplace_back(readID);
```
Comment (paoloczi) [inline src/AssemblerHttpServer-ReadGraph.cpp:42]: I would not bother with `emplace_back` here and use the more standard `push_back` instead. Unless I misunderstand something, it provides no benefits for this code where you are just dealing with integers.

File: src/AssemblerHttpServer-ReadGraph.cpp (hunk 2 above)
Comment (paoloczi) [reply src/AssemblerHttpServer-ReadGraph.cpp:34]: The same comment also in a few more places below.

File: src/AssemblerReadGraph.cpp [hunk 4]
```diff
@@ -161,116 +161,142 @@ void Assembler::checkReadGraphIsOpen()
...
-    q.push(start);
+        // Initialize a BFS starting at the start vertex.
+        std::queue<OrientedReadId> q;
+        q.push(start);
 
 
 
-    // Do the BFS.
-    while(!q.empty()) {
+        // Do the BFS.
```
Comment (paoloczi) [inline src/AssemblerReadGraph.cpp:221]: Wouldn't it be better to do a single BFS, initializing the queue with all of the start vertices?

File: src/AssemblerHttpServer-ReadGraph.cpp (hunk 3 above)
Comment (rlorigro) [reply src/AssemblerHttpServer-ReadGraph.cpp:42]: As written, I dont think it would provide any benefit. If instead I did:
`readIds.emplace_back(token);`

It would avoid copying or moving the `readId` object. Would you be ok with this implementation?

Comment (paoloczi) [reply src/AssemblerHttpServer-ReadGraph.cpp:42]: No, the compiler would optimize all of that away anyway. I think `emplace_back` is meant for situations where there is a significant cost in object construction, and/or object construction is complex enough that the compiler cannot do a good job inlining it.

And more in general, in areas of the code that are not important for performance, we should avoid optimizations that make the code less readable.

File: src/AssemblerReadGraph.cpp [hunk 5]
```diff
@@ -207,32 +235,32 @@ bool Assembler::createLocalReadGraph(
         const uint32_t distance1 = distance0 + 1;
 
         // Loop over edges of the global read graph involving this vertex.
-        for(const uint64_t i: readGraph.connectivity[orientedReadId0.getValue()]) {
+        for (const uint64_t i: readGraph.connectivity[orientedReadId0.getValue()]) {
             SHASTA_ASSERT(i < readGraph.edges.size());
-            const ReadGraphEdge& globalEdge = readGraph.edges[i];
+            const ReadGraphEdge &globalEdge = readGraph.edges[i];
```
Comment (paoloczi) [inline src/AssemblerReadGraph.cpp:240]: For consistency with the Shasta standard style and common C++ usage, please put this back as it was, that is, change to the following (space after the `&`, not before):

```
const ReadGraphEdge& globalEdge
```
There is at least one more place below (when defining `alignment`) where a similar change 
 back to the original code is needed.

File: src/AssemblerReadGraph.cpp [hunk 6]
```diff
@@ -207,32 +235,32 @@ bool Assembler::createLocalReadGraph(
...
             }
 
             // Get alignment information.
-            const AlignmentData& alignment = alignmentData[globalEdge.alignmentId];
+            const AlignmentData &alignment = alignmentData[globalEdge.alignmentId];
             OrientedReadId alignmentOrientedReadId0(alignment.readIds[0], 0);
             OrientedReadId alignmentOrientedReadId1(alignment.readIds[1], alignment.isSameStrand ? 0 : 1);
             AlignmentInfo alignmentInfo = alignment.info;
-            if(alignmentOrientedReadId0.getReadId() != orientedReadId0.getReadId()) {
+            if (alignmentOrientedReadId0.getReadId() != orientedReadId0.getReadId()) {
```
Comment (paoloczi) [inline src/AssemblerReadGraph.cpp:259]: I am not necessarily against a space  preceding parentheses after all `if`, but unrelated formatting changes make PRs harder to review. No need to change them back.

File: src/AssemblerReadGraph.cpp (hunk 5 above)
Comment (rlorigro) [reply src/AssemblerReadGraph.cpp:240]: Yes, my mistake, I actually prefer the `&` to be part of the type as well. It must have been swapped by my IDE

File: src/AssemblerReadGraph.cpp (hunk 6 above)
Comment (rlorigro) [reply src/AssemblerReadGraph.cpp:259]: I think this is another case of the IDE changing things
